Gem/Code/Source/Components/Multiplayer/GemSpawnerComponent.cpp

```diff
@@ -208,13 +209,14 @@ namespace MultiplayerSample
                     {
                         gemController->SetRandomPeriodOffset(GetNetworkRandomComponentController()->GetRandomInt() % 1000);
                         gemController->SetGemScoreValue(spawnable->m_scoreValue);
+                        gemController->SetGemSpawnerController(this, ticketId);
                     }
                 }
             }
 
             // Save the gem spawn ticket, otherwise the gem will immediately despawn due to the ticket's destruction.
             // Also track the root entity id so that we can move the gem out of sight while waiting for it to despawn when removing gems.
-            m_spawnedGems.emplace_back(AZStd::move(ticket), rootEntityId);
+            m_spawnedGems.insert(AZStd::make_pair(ticketId, SpawnedGem{ AZStd::move(ticket), rootEntityId }));
```

Comment:
Since you're removing the translation(-1000), we don't need to store the rootEntityId any more, you can just store ticketId and ticket.

Reply:
Sounds good.

Reply:
Done.